README.md

```diff
@@ -68,6 +68,20 @@ Once you have your dependencies configured, you can import packages using your i
 import "@openzeppelin/token/ERC721/ERC721.sol";
 ```
 
+### Library Linking
+
+To link libraries, add them to your `ape-config.yaml` file:
+
+```yaml
+solidity:
+  libraries:
+    subfolder/MySolidityFile.sol:
```

Comment:
I want to do:

```
solidity:
  libraries:
    ethereum:
      mainnet:
        SourceId.sol:
          MyLib: Address
```

but `ape compile` does not use the provider and there is no way of knowing which network deployment to use.

I was thinking of adding `--network` flag to `ape compile`.. It wouldn't connect to the network, but compiler plugins could use the network option to know which compile-settings to use.

Reply:
@fubuloubu Any thoughts on how to make this more network-friendly?

Reply:
When calling `ape test`, it executes a compile under the hood, so in that context it would know what network to deploy to

Could look up using the previous deployments for the project. If one exists, pull address from there for deploying, otherwise say you need the Lib deployed first.

I don't think compiling with the placeholder bytecode is a problem, just when you go to deploy it should first ensure that you have fully-formed bytecode, and raise an error if it doesn't

The manifest format is supposed to have a linking structure it in, but I don't know how it works well enough to advocate for it's use, and think it would be better just to roll with the weird way Solidity handles it

Reply:
Ha, when I originally wrote this, I didn't realize we were focused on local / testing

OK, I ❤️ the idea of using the deployments to automatically build the libraries here. The issue I can't piece together though is how that would happen automatically.

So I added a method `set_library()` that one could call after they deploy one of their libs and a test showing how to do that... Still thinking though also.

```python
library = project.Set.deploy(sender=account)
solidity = project.compiler_manager.registered_compilers[".sol"]
solidity.set_library(library)
```

Also, IICR, Pydantic validators from `ethpm-types` don't like the library stuff in the bytecode right now.

Reply:
Would be:
1. widening the validators to allow linking formatting
2. search for linking formatting within bytecode when you go and deploy it
3. if you see a link, try and make it based on your deployments, otherwise raise
4. deploy normally

But can definitely see how that's gonna take a some changes to `ethpm-types`, etc. to put together (don't think it's breaking change though)